### ISSUE TITLE
Fake rotation?

### ISSUE THREAD

**Issue by etodd** [MakeSchool-Tutorials/Hoppy-Bunny-SpriteKit-Swift#24]
Just throwing out an idea here.

Using the physics engine for bunny rotation seems overly complicated, and gives some weird results, especially when the bunny starts rolling on the ground. It also requires the CGFloat clamping code and a touch timer.

Here's what I propose:

- Have them uncheck the "Allow rotation" checkbox on the bunny's physics body
- Do the scrolling first, get the bunny moving forward
- Skip the touch timer, angular impulse, fixedDelta, and CGFloat clamping
- Instead of all that, have them write a line like this in the update function:
```swift
hero.zRotation = CGFloat(atan2f(scrollSpeed, -Float(hero.physicsBody!.velocity.dy)) + M_PI * -0.5)
```
It just makes the bunny face toward its velocity. It can't hurt to teach people to use atan2f to make stuff face a certain direction. It's one of the most common questions people have.

Just a thought. If it seems like a good idea I'd be happy to make a pull request.

**Comment by dionlarson** [comment]
I believe changing the rotation manually on hero will still update the physics engine.

With the collision mask I added (during game over) I believe it should no longer have too much of a rolling issue anymore?

**Comment by etodd** [comment]
Yeah. The physics engine gets updated, but with allowRotation set to false, collisions no longer cause the body to rotate weirdly.

I encountered the rolling issue in P4 before we apply any collision masks. It isn't a big deal at all, feel free to close this. Just thought it might be an opportunity to simplify things.

**Comment by dionlarson** [comment]
Yeah. I'm not entirely against it so I'm going to leave this open for now. That said, changing it is not a priority right now since this is not a breaking issue.
